crypto/params_from_text.c

```diff
@@ -152,30 +138,45 @@ static int construct_from_text(OSSL_PARAM *to, const OSSL_PARAM *paramdef,
         }
     }
 
-    *to = *paramdef;
+    to->key = template->key;
+    to->data_type = template->data_type;
     to->data = buf;
     to->data_size = buf_n;
     to->return_size = 0;
 
     return 1;
 }
 
-int OSSL_PARAM_construct_from_text(OSSL_PARAM *to,
-                                   const OSSL_PARAM *paramdefs,
-                                   const char *key, const char *value,
-                                   size_t value_n,
+#define FLAG_ISHEX             1
+
+const OSSL_PARAM *OSSL_PARAM_parse_locate_const(const OSSL_PARAM *params,
+                                                const char *key,
+                                                int *flags)
```

Comment:
`unsigned int` for flags perhaps?  We usually avoid signed flags.

Reply:
Done